nvaccess/nvda
# Windows 11 Notepad: take 2 on announcing status bar contents in newer app releases via UIA decendants

### Link to issue number:
Closes #14573
Take 2 on #14573
Resolves a flaw discovered in #14574 

### Summary of the issue:
In #14574 and earlier, NVDA relied on child index when retrieving Windows 11 Notepad status bar contents. This is flawed: there are times when UI tree in Notepad can change if something opens and then closes. Specifically, when Notepad settings window (Alt+S) is opened and then closed, NVDA will announce something else when retrieving status bar contents.

### Description of user facing changes
NVDA will announce status bar contents in newer Windows 11 Notepad releases (same as #14574).

### Description of development approach
Use UIA's FindFirstBuildCache to retrieve status bar contents (Automation Id: ContentTextBlock). As long as the Automation Id stays the same, the descendants approach will work across Notepad 11 releases.

### Testing strategy:
Manual testing (prerequisites: multiple Windows 11 computers, one being part of Windows Insidr Preview):

1. Open Notepad.
2. Enable status bar (View/Status bar is checked).
3. Type something.
4. Press status bar command (NVDA+End in desktop layout, NVDA+Shift+End in laptop layout).
5. In Notepad 11.2301 and later, press Control+N to open a new tab.
6. Perform status bar announcement command.
7. Press Alt+S to open Notepad settings window.
8. Press Backspace to close settings window.
9. Perform status bar announcement command again.

Expected: NVDA announces status bar contents such as character positoin and encoding.
Prior to #14574: NVDA will not nanounce status bar contents or say something else in Notepad 11.2301 and later.
With #14574: NVDA will sya something else when trying to announce the status bar after opening and closing Notepad settings.
With take 2: NVDA will announce status bar contents across Notepad 11 releases (with or without tabs and opening and closing settings window).

### Known issues with pull request:
None

### Change log entries:
Same as #14574 

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
